pkg/controller/jobframework/integrationmanager_test.go

```diff
@@ -312,38 +416,45 @@ func TestGetCallbacksForOwner(t *testing.T) {
 			"manageK1":   manageK1,
 			"manageK2":   manageK2,
 		},
+		externalIntegrations: map[string]runtime.Object{
+			"externalK3": externalK3,
+		},
 	}
 
 	cases := map[string]struct {
-		owner         *metav1.OwnerReference
-		wantCallbacks *IntegrationCallbacks
+		owner       *metav1.OwnerReference
+		wantJobType runtime.Object
 	}{
 		"K1": {
-			owner:         &metav1.OwnerReference{Kind: "K1"},
-			wantCallbacks: &manageK1,
+			owner:       &metav1.OwnerReference{Kind: "K1"},
+			wantJobType: manageK1.JobType,
 		},
 		"K2": {
-			owner:         &metav1.OwnerReference{Kind: "K2"},
-			wantCallbacks: &manageK2,
+			owner:       &metav1.OwnerReference{Kind: "K2"},
+			wantJobType: manageK2.JobType,
 		},
 		"K3": {
-			owner:         &metav1.OwnerReference{Kind: "K3"},
-			wantCallbacks: nil,
+			owner:       &metav1.OwnerReference{Kind: "K3"},
+			wantJobType: externalK3,
+		},
+		"K4": {
+			owner:       &metav1.OwnerReference{Kind: "K4"},
+			wantJobType: nil,
 		},
 	}
 
 	for tcName, tc := range cases {
 		t.Run(tcName, func(t *testing.T) {
-			gotCallbacks := mgr.getCallbacksForOwner(tc.owner)
-			if tc.wantCallbacks == nil {
-				if gotCallbacks != nil {
+			wantJobType := mgr.getJobTypeForOwner(tc.owner)
+			if tc.wantJobType == nil {
+				if wantJobType != nil {
 					t.Errorf("This owner should be unmanaged")
 				}
 			} else {
-				if gotCallbacks == nil {
+				if wantJobType == nil {
 					t.Errorf("This owner should be managed")
 				} else {
-					if diff := cmp.Diff(*tc.wantCallbacks, *gotCallbacks, cmp.FilterValues(func(_, _ interface{}) bool { return true }, cmp.Comparer(compareCallbacks))); diff != "" {
+					if diff := cmp.Diff(tc.wantJobType, wantJobType, cmp.FilterValues(func(_, _ interface{}) bool { return true }, cmp.Comparer(compareJobTypes))); diff != "" {
```

Comment:
Are you sure you need a custom comparer after this change?

Reply:
Not needed; simplified.